frontend/src/metabase/collections/utils.ts

```diff
@@ -11,15 +11,27 @@ export function nonPersonalOrArchivedCollection(
   collection: Collection,
 ): boolean {
   // @TODO - should this be an API thing?
-  return !isPersonalCollection(collection) && !collection.archived;
+  return !isRootPersonalCollection(collection) && !collection.archived;
 }
 
-export function isPersonalCollection(
+export function isRootPersonalCollection(
   collection: Partial<Collection> | CollectionItem,
 ): boolean {
   return typeof collection.personal_owner_id === "number";
 }
 
+export function isPersonalCollection(
+  collection: Pick<Collection, "is_personal">,
```

Comment:
We want to allow passing `ROOT_COLLECTION` directly and since it doesn't contain all the necessary properties in `Collection`, we can't use `Collection` directly.

And this util only care if an object contains `is_personal`, so this type should be enough.